src/Filters/WhereNull.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/*
+ * Copyright 2022 Cloud Creativity Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace LaravelJsonApi\Eloquent\Filters;
+
+use LaravelJsonApi\Core\Support\Str;
+use LaravelJsonApi\Eloquent\Contracts\Filter;
+
+class WhereNull implements Filter
+{
+    use Concerns\HasColumn;
+    use Concerns\IsSingular;
+
+    /**
+     * @var bool
+     */
+    private bool $not;
+
+    /**
+     * Create a new filter.
+     *
+     * @param string $key
+     * @param bool $not
+     * @param string|null $column
+     * @return static
+     */
+    public static function make(string $key, bool $not, string $column = null): self
+    {
+        return new static($key, $not, $column);
+    }
+
+    /**
+     * WhereNotNull constructor.
+     *
+     * @param string|null $column
+     */
+    public function __construct(string $key, bool $not, string $column = null)
+    {
+        $this->key = $key;
+        $this->column = $column ?: $this->guessColumn();
+        $this->not = $not;
+    }
+
+    public function key(): string
+    {
+        return $this->key;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function apply($query, $value)
+    {
+        return $query->whereNull($this->key,'and',$this->not);
```

Comment:
You should use (and qualify) the column here instead of the key.

Reply:
Agree with this, the column name needs to be used and it needs to be qualified as this could be used within a relationship.